ac/ac-brainstorm/src/Dashboard.jsx

```diff
@@ -3,48 +3,46 @@
 import React from 'react';
 import { Chart } from 'react-google-charts';
 
-export default ({ logs }: Object) => {
-  const data = {};
-  logs.forEach(log => {
-    data[log.data.type] = data[log.data.type] || {};
-    data[log.data.type][log.data.key] =
-      (data[log.data.type][log.data.key] || 0) + 1;
-  });
+const Viewer = ({ state }: { state: Object }) => (
+  <div>
+    {state.idea && (
+      <Chart
+        chartType="BarChart"
+        data={[
+          ['Name', 'Count'],
+          ...Object.keys(state.idea).map(key => [key, state.idea[key]])
+        ]}
+        width="100%"
+        height="300px"
+        options={{
+          title: 'number of suggested ideas per group',
+          bar: { groupWidth: '95%' },
+          legend: { position: 'none' }
+        }}
+      />
+    )}
+    {state.vote && (
+      <Chart
+        chartType="BarChart"
+        data={[
+          ['Name', 'Count'],
+          ...Object.keys(state.vote).map(key => [key, state.vote[key]])
+        ]}
+        width="100%"
+        height="300px"
+        options={{
+          title: 'number of votes per student',
+          bar: { groupWidth: '95%' },
+          legend: { position: 'none' }
+        }}
+      />
+    )}
+  </div>
+);
 
-  return (
-    <div>
-      {data.idea && (
-        <Chart
-          chartType="BarChart"
-          data={[
-            ['Name', 'Count'],
-            ...Object.keys(data.idea).map(key => [key, data.idea[key]])
-          ]}
-          width="100%"
-          height="300px"
-          options={{
-            title: 'number of suggested ideas per group',
-            bar: { groupWidth: '95%' },
-            legend: { position: 'none' }
-          }}
-        />
-      )}
-      {data.vote && (
-        <Chart
-          chartType="BarChart"
-          data={[
-            ['Name', 'Count'],
-            ...Object.keys(data.vote).map(key => [key, data.vote[key]])
-          ]}
-          width="100%"
-          height="300px"
-          options={{
-            title: 'number of votes per student',
-            bar: { groupWidth: '95%' },
-            legend: { position: 'none' }
-          }}
-        />
-      )}
-    </div>
-  );
+const mergeLog = (state: Object, log: Object) => {
+  state[log.type] = state[log.type] || {};
+  state[log.type][log.instanceId] = (state[log.type][log.instanceId] || 0) + 1;
 };
+
+export default { 'Event types': { initData: {}, mergeLog, Viewer } };
```

Comment:
I feel that the code is less messy with the new API. I like how this dashboard here looks very reusable: counting eventType per instance